source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -257,6 +258,14 @@ FilterStatus Router::messageBegin(MessageMetadataSharedPtr metadata) {
 
   ENVOY_STREAM_LOG(debug, "router decoding request", *callbacks_);
 
+  if (route_entry_->stripServiceName()) {
+    const auto& method = metadata->methodName();
+    const auto pos = method.find(':');
+    if (pos != std::string::npos) {
+      metadata->setMethodName(method.substr(pos + 1));
```

Comment:
Do we need to concerned with the method name becoming an empty string?

Reply:
you mean `foo:` -> ``? Hmm, have you ever seen method names with colons for something else then prefixing the service name?

cc: @fishcakez

Reply:
I don't expect it to be common, but if it causes a crash it can be turned into an attack vector. If it just produces an invalid request that upstreams would simply reject, it's probably fine.

Reply:
Ah -- it won't crash: https://godbolt.org/z/KNVg64

Reply:
Oh, hmm you probably meant somewhere else in the code we might expect this to be non empty...

Reply:
We set an empty method name here:

https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/network/thrift_proxy/binary_protocol_impl.cc#L51

and in a few other places, so it's probably fine..

Reply:
A thrift identifier is required to be of the form `[a-zA-Z_](\.[a-zA-Z_0-9]|[a-zA-Z_0-9])*` so method name should never contain `:`. However therefore it's also invalid to send an empty method.

The Thrift protocols supported by envoy prefix the method name string with the length of the method name string. Im unsure how that might get handled if the string is empty in the different language implementations.

Reply:
Alright. I don't see anywhere that we assert on the method name being non-empty. So I think it's ok from Envoy's perspective.